### PR TITLE
Update analytics.js to gtag to support ga4

### PR DESCRIPTION
**Pull request by martoboto**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌
| New feature/enhancement? (use the a.x branch)      | ✅
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ✅
| Automated tests included?              | ✅ 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

This is the same as https://github.com/mautic/mautic/pull/12421 but on the v4 codebase.

#### Description:

Google will turn off universal analytics on July 1st and the new GA4 codes do not work with analytics.js. This PR upgrades the analytics.js code to use the newer gtag code which will work with old UA codes and the newer GA4 codes.

This is a breaking change if the frontend uses analytics.js for goal or ecommerce tracking the end user would need to update that code to gtag or lose that reporting.

#### Steps to test this PR:

Create a new GA4 analytics account and get the code.
In mautic under Configuration > Tracking Settings > Google Analytics
Paste in your Google Analytics ID and enable on tracking page and landing page.
Load a landing page in a private tab.
Check analytics real time you will see your page hit.

### CODE REVIEW

**Comment by RCheesley** [comment]
Thanks for making the PR @martoboto 🚀

File: app/bundles/PageBundle/EventListener/BuildJsSubscriber.php [hunk 1]
```diff
@@ -560,13 +572,12 @@ public function onBuildJsForTrackingEvent(BuildJsEvent $event)
...
                      for(var i = 0; i < e.length; i++) {
-                         if(typeof e[i]['action']  !== 'undefined' && typeof e[i]['label']  !== 'undefined' )
-                             	ga('send', {
-                                    hitType: 'event',
-                                    eventCategory: e[i]['category'],
-                                    eventAction: e[i]['action'],
-                                    eventLabel: e[i]['label'],
-			                     });
+                         if(typeof e[i]['action']  !== 'undefined' && typeof e[i]['label']  !== 'undefined' ) {
+                            gtag('event', e[i]['action'], {
```
Comment (escopecz) [inline app/bundles/PageBundle/EventListener/BuildJsSubscriber.php:583]: This change is missing from https://github.com/mautic/mautic/pull/12421. Is that on purpose?

Comment (martoboto) [reply app/bundles/PageBundle/EventListener/BuildJsSubscriber.php:583]: Nope I just noticed that the facebook code uses analytics.js. Ill need to add this to the v5 pr too and test it properly.

**Comment by RCheesley** [comment]
Sorry to be a pain but this needs to be rebased to 4.4, as that's where we make the patch releases from, rather than 4.x!

**Comment by martoboto** [comment]
My bad. Do I need to change the PR to be to 4.4 too?

**Comment by escopecz** [comment]
@martoboto yes. I just did that. Thank you!

File: app/bundles/PageBundle/EventListener/BuildJsSubscriber.php (hunk 1 above)
Comment (martoboto) [reply app/bundles/PageBundle/EventListener/BuildJsSubscriber.php:583]: I have updated the analytics.js event to gtag for the v5 one too and tested it locally.

**Review by escopecz — APPROVED** [review]
I tested this with a landing page and GA v4 tracking key and it worked 👍 

Screenshot 2023-07-18 at 13 17 38

I'm merging this without a second approval as this should have been released a month ago when the GA Universal stopped working.

**Pull request merged**